indexing-service/src/main/java/io/druid/indexing/overlord/TaskLockbox.java

```
@@ -210,12 +216,16 @@ public TaskLock lock(final Task task, final Interval interval) throws Interrupte
    * @param preferredVersion use this version string if one has not yet been assigned
    *
    * @return lock version if lock was acquired, absent otherwise
+   * @throws IllegalStateException if the task is not a valid active task
    */
   private Optional<TaskLock> tryLock(final Task task, final Interval interval, final Optional<String> preferredVersion)
   {
     giant.lock();
 
     try {
+      if(!activeTasks.contains(task.getId())){
```

Why throw ISE instead of returning absent?